### PR TITLE
Add SQLite Wasm article

### PR DESCRIPTION
**Pull request by tomayac**
Changes proposed in this pull request:

- Add the article.
- Preview: https://deploy-preview-4620--developer-chrome-com.netlify.app/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/

### CODE REVIEW

**Comment by tomayac** [comment]
Reviewed by @rachelandrew in the corresponding [doc](https://docs.google.com/document/d/1Hjh0DgvcBwigAkkWtNFxj0Wg8kqOtEt5I6ETgYFmGcw/edit?pli=1&resourcekey=0-abyBxI9euDWhdgvIXjbTeg). This is to go live on Jan 11, 2023. `DO NOT MERGE` until then. Thanks all!

File: site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+The Origin Private File System (OPFS, part of the
+[File System Access API](/articles/file-system-access/)) is augmented with a special surface that
+brings very performant access to data. This new surface differs from existing ones by offering
+in-place and exclusive write access to a file's content. This change, along with the ability to
+consistently read unflushed modifications and the availability of a synchronous variant on dedicated
+workers, significantly improves performance and unblocks new use cases.
+
+As you can imagine, the last point of the project's goals, _Support for persistent client-side
+storage using available JavaScript APIs_ specifically, comes with strict performance requirements
```
Comment (rachelandrew) [inline site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md:73]:
```suggestion
As you can imagine, the last point of the project's goals, Support for persistent client-side
storage using available JavaScript APIs, comes with strict performance requirements
```

File: site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+objects comes into play. This method returns a Promise which resolves to a
+[`FileSystemSyncAccessHandle`](https://developer.mozilla.org/docs/Web/API/FileSystemSyncAccessHandle)
+object that can be used to synchronously read from and write to a file. The synchronous nature of
+this method brings performance advantages, but therefore it is only usable inside dedicated
+[Web Workers](https://developer.mozilla.org/docs/Web/API/Web_Workers_API) for files within the
+Origin Private File System so the main thread can't be blocked.
+
+### Setting the required headers
+
+{% Aside 'warning' %} Setting these headers is a _required_ step for the Origin Private File System
```
Comment (rachelandrew) [inline site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md:87]:
```suggestion
{% Aside 'warning' %} Setting these headers is a required step for the Origin Private File System
```

File: site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+the user's device, making it easy to integrate into applications.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/eDfwxFJirAZ6hqdt6Z8Y.png", alt="SQLite logo.", width="220", height="101" %}
+
+## SQLite based on Web Assembly
+
+There are a number of unofficial SQLite versions based on Web Assembly (Wasm), allowing it to be
+used in web browsers, for example, [sql.js](https://github.com/sql-js/sql.js). The
+[sqlite3 WASM/JS subproject](https://sqlite.org/wasm/doc/tip/about.md) is the first effort that is
+"officially" associated with the [SQLite project](https://sqlite.org/) makingWasm builds of the
```
Comment (rachelandrew) [inline site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md:38]:
```suggestion
"officially" associated with the [SQLite project](https://sqlite.org/) making Wasm builds of the
```

Comment (rachelandrew) [reply site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md:38]: also why the scare quotes round officially?

**Review by rachelandrew — CHANGES_REQUESTED** [review]

File: site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md (hunk 3 above)
Comment (tomayac) [reply site/en/blog/sqlite-wasm-in-the-browser-backed-by-the-origin-private-file-system/index.md:38]: Quotes be gone.

**Review by rachelandrew — APPROVED** [review]

**Comment by ankurpiparsania** [comment]
Below code throws error that sqlite3.opfs doesnt have OpfsDb
const capi = sqlite3.capi; /*C-style API*/
const oo = sqlite3.oo1; /*high-level OO API*/
if (sqlite3.opfs) {
    db = new sqlite3.opfs.OpfsDb('/mydb.sqlite3');  

// ERROR: sqlite3.opfs doesnt have OpfsDb 
//FIX: db = new oo.OpfsDb('/mydb.sqlite3')

    log('The OPFS is available.');
  } else {
    db = new oo.DB('/mydb.sqlite3', 'ct');
    log('The OPFS is not available.');
  }

**Comment by tomayac** [comment]
This was changed in https://sqlite.org/src/info/0c5c51f4. Check out https://stackblitz.com/edit/vitejs-vite-ttrbwh?file=worker.js for a working version that uses the latest API.

**Comment by tomayac** [comment]
@ankurpiparsania Sorry, I read part of your comment wrong. I have opened https://github.com/GoogleChrome/developer.chrome.com/pull/7097 that fixes the issue you reported.

**Pull request merged**